aclk/aclk_common.h

```diff
@@ -31,6 +49,11 @@ extern struct aclk_shared_state {
     ACLK_METADATA_STATE metadata_submitted;
     ACLK_AGENT_STATE agent_state;
     time_t last_popcorn_interrupt;
+
+    // read only while ACLK connected
+    // protect by lock otherwise
+    volatile int version_neg;
```

Comment:
I don't think that volatile is needed here as access is controlled by a lock.

Reply:
yes thank you, I think you are right that volatile is not needed in the final solution.
it is protected by lock during the initialization phase (where we can write to it). Afterward (when ACLK is running) read-only access is allowed and thus lock is not necessary.